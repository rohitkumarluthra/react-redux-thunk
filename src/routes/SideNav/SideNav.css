.side {
    flex: 20%;
    /* background-color: #bbb8b8; */
    padding: 20px;
    height: 100vh;
}

.navbar {
    display: flex;
}

.navbar li {
  padding: 10px;
  width: 100%;
}

.navbar li:hover {
    cursor: pointer;
    background-color: #ddd;
    color: black;
}

.link {
  text-decoration: none;
}

.active_link {
  background-color: #ddd;
  color: #0000AA;
}