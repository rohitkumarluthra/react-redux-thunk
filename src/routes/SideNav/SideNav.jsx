import React from 'react';
import { connect } from "react-redux";

import "./SideNav.css";
import { selectCategory, fetchCategories } from '../../redux/categories/actions';

export class SideNav extends React.PureComponent {

    render() {
      const { items } = this.props.categories;

      return (
          <div className="side">
            <h2>Categories</h2>
            <div className="navbar">
              <ul>                  
                {this.renderCategories(items)}
              </ul>
            </div>
          </div>
      );
    }

    componentDidMount() {
      this.props.getAllCategories();
    }

    ///////////////////////////////////////////////////////////////////////
    //  RENDER METHODS
    ///////////////////////////////////////////////////////////////////////
    renderCategories(categories) {
      const { activeCategory } = this.props.categories;

      return categories.map((category, i) => {
        let style = (activeCategory === category.id) ? 'active_link' : '';
        return (<li className={style} key={category.id} onClick={() => this.handleSelectCategory(category.id)}>{category.name}</li>)
      })
    }

    ///////////////////////////////////////////////////////////////////////
    //  EVENT HANDLERS
    ///////////////////////////////////////////////////////////////////////
    handleSelectCategory = (categoryId) => {
      this.props.selectCategory(categoryId)
    }

}

///////////////////////////////////////////////////////////////////////
//  REDUX CONNECTION
///////////////////////////////////////////////////////////////////////
function mapDispatchToProps(dispatch) {
  return {
    getAllCategories: () => dispatch(fetchCategories()),
    selectCategory: (categoryId) => dispatch(selectCategory(categoryId))
  };
}

function mapStateToProps(state) {
  const { categories } = state;
  const categoriesToJS = categories.toJS();

  return {
    categories : categoriesToJS.categories,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);