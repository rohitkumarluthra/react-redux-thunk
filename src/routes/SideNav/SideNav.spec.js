import React from 'react';
import { act } from 'react-dom/test-utils';
import SideNav from './SideNav';
import '../../setupTests';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import { shallow, mount } from 'enzyme';


describe('SideNav renders',() => {
  it('renders', () => {
    const wrapper = shallow(<SideNav store={store} />);
    expect(wrapper.exists()).toBe(true);
  })

  it('SideNav component created', () => {
    let rendered;
    act(() => {
      rendered = renderer.create(
        <SideNav store={store} />
      );
      expect(rendered.toJSON()).toMatchSnapshot();
    });
  })

  it('find text:Categories', () => {
    const wrapper = mount(<SideNav store={store} />);
    const text = wrapper.find('h2').text();
    expect(text).toEqual('Categories');
  });

  it('side nav bar ul count', () => {
    const wrapper = mount(<SideNav store={store} />);
    const navBarLink = wrapper.find('ul');
    expect(navBarLink.length).toEqual(1);
  });
})