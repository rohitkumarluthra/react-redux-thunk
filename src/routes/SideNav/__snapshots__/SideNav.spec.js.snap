// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders SideNav component created 1`] = `
<div
  className="side"
>
  <h2>
    Categories
  </h2>
  <div
    className="navbar"
  >
    <ul />
  </div>
</div>
`;
